File: apps/api/src/main/java/com/labelhub/entity/AgentRun.java
package com.labelhub.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "agent_run")
@Data
public class AgentRun {

    @Id
    private String id;

    @Column(nullable = false)
    private String agentName;

    @Column(nullable = false)
    private String skillName;

    @Column(nullable = false)
    private String skillVersion;

    @Column(nullable = false)
    private String taskId;

    private String itemId;

    @Column(nullable = false)
    private String traceId;

    @Column(nullable = false)
    private String status;

    @Column(columnDefinition = "TEXT")
    private String inputSummary;

    @Column(columnDefinition = "TEXT")
    private String outputSummary;

    @Column(columnDefinition = "TEXT")
    private String contextRefsJson;

    @Column(columnDefinition = "TEXT")
    private String toolCallsJson;

    @Column(columnDefinition = "TEXT")
    private String llmCallsJson;

    private Long latencyMs;

    private Long queueWaitMs;

    private String errorCode;

    @Column(columnDefinition = "TEXT")
    private String errorMessage;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime completedAt;
}

File: apps/agent-runtime/agent_runtime/schemas/agent_run.py
from pydantic import BaseModel, Field
from typing import Optional, Any, Dict, List
from datetime import datetime


class ToolCallRecord(BaseModel):
    tool_name: str
    arguments_summary: str
    result_summary: str
    latency_ms: int
    error_code: Optional[str] = None


class LlmCallRecord(BaseModel):
    model_name: str
    prompt_version: str
    input_tokens: int
    output_tokens: int
    latency_ms: int
    validation_passed: bool


class AgentRunRecord(BaseModel):
    id: str
    agent_name: str
    skill_name: str
    skill_version: str
    task_id: str
    item_id: Optional[str] = None
    trace_id: str
    status: str
    input_summary: str
    output_summary: str
    context_refs_json: str
    tool_calls_json: str
    llm_calls_json: str
    latency_ms: Optional[int] = None
    queue_wait_ms: Optional[int] = None
    error_code: Optional[str] = None
    error_message: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None


class AgentResultOutput(BaseModel):
    id: str
    agent_run_id: str
    task_id: str
    item_id: str
    result_type: str
    suggestion: str
    confidence: float
    evidence_json: str
    metadata_json: str
    status: str
    created_at: datetime
